ts/IterableInboxMessageCell.tsx

```diff
@@ -1,47 +1,111 @@
             <View style={styles.unreadMessageContainer}>
+                  <Text style={styles.title}>{messageTitle}</Text>
+                  <Text style={styles.body}>{messageBody}</Text>
+                  <Text style={styles.timestamp}>{messageCreatedAt}</Text>
+            </View>     
+         </TouchableOpacity>)    
+   }
+
+   function displayReadMessage() {
+      return (
+         <TouchableOpacity
+            style={(active) ? styles.pressedMessageCell : styles.messageCell}
+            activeOpacity={1}
+            onPress={() => {
+               setActive(!active);
+               handleMessageSelect(index);
+            }}
+         >
+            <View style={styles.readMessageContainer}>
                <Text style={styles.title}>{messageTitle}</Text>
                <Text style={styles.body}>{messageBody}</Text>
                <Text style={styles.timestamp}>{messageCreatedAt}</Text>
-            </View>      
-         </View>)    
+            </View>  
+         </TouchableOpacity>)    
    }
 
-   function displayReadMessage() {
-      return ( 
-         <View style={last ? styles.lastMessageCell : styles.messageCell}>
+   function displayUnreadLastMessage() {
+      return (
+         <TouchableOpacity
+            style={(active) ? styles.pressedMessageCell : styles.messageCell}
+            activeOpacity={1}
+            onPress={() => {
+               setActive(!active);
+               handleMessageSelect(index);
+            }}
+         >
+            <View style={styles.unreadIndicatorContainer}>
+               <View style={styles.unreadIndicator}/>
+            </View>
+            <View style={styles.unreadMessageContainer}>
+               <Text style={styles.title}>{messageTitle}</Text>
+               <Text style={styles.body}>{messageBody}</Text>
+               <Text style={styles.timestamp}>{messageCreatedAt}</Text>
+            </View>   
+         </TouchableOpacity>)    
+   }
+
+   function displayReadLastMessage() {
+      return (
+         <TouchableOpacity
+            style={(active) ? styles.pressedMessageCell : styles.messageCell}
+            activeOpacity={1}
+            onPress={() => {
+               setActive(!active);
+               handleMessageSelect(index);
+            }}
+         >
             <View style={styles.readMessageContainer}>
                <Text style={styles.title}>{messageTitle}</Text>
                <Text style={styles.body}>{messageBody}</Text>
                <Text style={styles.timestamp}>{messageCreatedAt}</Text>
-            </View>      
-         </View>)    
+            </View> 
+         </TouchableOpacity>)    
    }
 
    return(
-      message.read ?
-         displayReadMessage() :
-         displayUnreadMessage()
+      <View>
+         {last ? 
+            (message.read ? displayReadLastMessage() : displayUnreadLastMessage()) :
+            (message.read ? displayReadMessage() : displayUnreadMessage())}
+      </View>   
```

Comment:
this is a little harder to follow than necessary, IMO. My general rule here is that: if whoever is on-call at 3am in the morning is just woken up and has to look at the code for the first time in their lives, can they easily parse what's happening?  I'd argue, a ternary inside of a ternary  is a bit complex. 

instead you could do something like: 

```
if (last) {
  return (
     <View>
       { message.read ? displayReadLastMessage() : displayUnreadLastMessage()) } 
     </View>
  )
} 

return (
      <View>
            { message.read ? displayReadMessage() : displayUnreadMessage())}
      </View>   
)
```

Reply:
Yes, I will make the appropriate changes.